contracts/cw-native-staked-balance-voting/src/contract.rs

```diff
@@ -0,0 +1,287 @@
+        &info.sender,
+        env.block.height,
+        |balance| -> StdResult<Uint128> { Ok(balance.unwrap_or_default().checked_add(amount)?) },
+    )?;
+    STAKED_TOTAL.update(
+        deps.storage,
+        env.block.height,
+        |total| -> StdResult<Uint128> { Ok(total.unwrap_or_default().checked_add(amount)?) },
+    )?;
+
+    Ok(Response::new()
+        .add_attribute("action", "stake")
+        .add_attribute("amount", amount.to_string())
+        .add_attribute("from", info.sender))
+}
+
+pub fn execute_unstake(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    amount: Uint128,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    STAKED_BALANCES.update(
+        deps.storage,
+        &info.sender,
+        env.block.height,
+        |balance| -> StdResult<Uint128> { Ok(balance.unwrap_or_default().checked_sub(amount)?) },
+    )?;
+    STAKED_TOTAL.update(
+        deps.storage,
+        env.block.height,
+        |total| -> StdResult<Uint128> { Ok(total.unwrap_or_default().checked_sub(amount)?) },
+    )?;
+
+    match config.unstaking_duration {
+        None => {
+            let msg = CosmosMsg::Bank(BankMsg::Send {
+                to_address: info.sender.to_string(),
+                amount: coins(amount.u128(), config.denom),
+            });
+            Ok(Response::new()
+                .add_message(msg)
+                .add_attribute("action", "unstake")
+                .add_attribute("from", info.sender)
+                .add_attribute("amount", amount)
+                .add_attribute("claim_duration", "None"))
+        }
+        Some(duration) => {
+            let outstanding_claims = CLAIMS.query_claims(deps.as_ref(), &info.sender)?.claims;
+            if outstanding_claims.len() >= MAX_CLAIMS as usize {
+                return Err(ContractError::TooManyClaims {});
+            }
+
+            CLAIMS.create_claim(
+                deps.storage,
+                &info.sender,
+                amount,
+                duration.after(&env.block),
+            )?;
+            Ok(Response::new()
+                .add_attribute("action", "unstake")
+                .add_attribute("from", info.sender)
+                .add_attribute("amount", amount)
+                .add_attribute("claim_duration", format!("{}", duration)))
+        }
+    }
+}
+
+pub fn execute_update_config(
+    deps: DepsMut,
+    info: MessageInfo,
+    new_owner: Option<String>,
+    new_manager: Option<String>,
+    duration: Option<Duration>,
+) -> Result<Response, ContractError> {
+    let new_owner = new_owner
+        .map(|new_owner| deps.api.addr_validate(&new_owner))
+        .transpose()?;
+    let new_manager = new_manager
+        .map(|new_manager| deps.api.addr_validate(&new_manager))
+        .transpose()?;
+    let mut config: Config = CONFIG.load(deps.storage)?;
+    if Some(info.sender.clone()) != config.owner && Some(info.sender.clone()) != config.manager {
```

Comment:
```suggestion
    if Some(info.sender) != config.owner || Some(info.sender) != config.manager {
```

also this check can be moved before the address validation of new owner / manager